cmd/aotutil/util.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+type WaitAction int
+
+const (
+	WaitContinue WaitAction = iota + 1
+	WaitDone
+)
+
+func Wait(interval, timeout time.Duration, f func() (WaitAction, error)) error {
+	// Run f once
+	act, err := f()
+	if err != nil {
+		return err
```

Comment:
I would prefer checking `act` as well before return in case of failure retries.

Reply:
it stops on error and we expect f to swallow the error if it thinks it is retry-able. I have updated the comment.